packages/toolpad-app/src/server/schema.ts

```diff
@@ -301,6 +308,7 @@ export const META = {
   definitions: {
     Json: jsonSchema,
     JsExpressionBinding: jsExpressionBindingSchema,
+    EnvBinding: envBindingSchema,
```

Comment:
An app created with this schema won't work in older Toolpad versions. Older Toolpad versions don't know what `$env` means

Reply:
The general idea would be:

- old schema + new toolpad => we should be able to internally migrate and display the app + give option for user to migrate files to new version, or do it automatically
- new schema + old toolpad => either all features of the schema are supported in Toolpad, or otherwise we need to tell the user "This app as created in newer Toolpad, please upgrade Toolpad to version X if you want to run/edit this application"

Reply:
At first thought the 2nd option seems more sane to me and not that unusual? 
But we can try the first one if it's not that hard, also depending on how common it would be for users to try to use new schemas in old Toolpad versions.

Reply:
These are two different scenarios that can happen, we have to deal with both of them

> also depending on how common it would be for users to try to use new schemas in old Toolpad versions.

This scenario could happen when people copy toolpad apps from one repo to another